File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int j = 0;
        int t = target;
        for(i = 0; i<nums.size();i++){
            for(int j = i+1; j<nums.size();j++){
            if(nums[i]+nums[j]==t){
                return {i,j};
            }
        }
        }
        return {};
    }
};

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.py
class Solution:
    def combinationSum2(self, candidates, target):
        candidates.sort()
        result = []

        def backtrack(start, current, total):
            if total == target:
                result.append(current[:])
                return

            if total > target:
                return

            for i in range(start, len(candidates)):

                # Duplicate combination avoid karo
                if i > start and candidates[i] == candidates[i - 1]:
                    continue

                current.append(candidates[i])

                # i + 1 because same element dobara use nahi kar sakte
                backtrack(i + 1, current, total + candidates[i])

                current.pop()

        backtrack(0, [], 0)
        return result
        